gunicorn/workers/base_async.py

```diff
@@ -107,14 +107,16 @@ def handle_request(self, listener_name, req, sock, addr):
             if self.is_already_handled(respiter):
                 return False
             try:
-                if isinstance(respiter, environ['wsgi.file_wrapper']):
-                    resp.write_file(respiter)
-                else:
-                    for item in respiter:
-                        resp.write(item)
-                resp.close()
-                request_time = datetime.now() - request_start
-                self.log.access(resp, req, environ, request_time)
+                try:
+                    if isinstance(respiter, environ['wsgi.file_wrapper']):
+                        resp.write_file(respiter)
+                    else:
+                        for item in respiter:
+                            resp.write(item)
+                    resp.close()
+                finally:
+                    request_time = datetime.now() - request_start
```

Comment:
why not put this in the `finally` below ? I don't see a need for wrapping it more than it is there

Reply:
I'm not so concerned about `request_time = datetime.now() - request_start` but I wasn't sure if `Logger.access` may fail, and I didn't want to introduce a new failure case where due to Logger failure, we then fail to call `respiter.close()`.

Looking at `Logger.access`, it suppresses `Exception` so it should be _safe enough_ to merge.